source/base/Client.cpp

```diff
@@ -0,0 +1,12 @@
+#include "Client.h"
+
+
+
+Client::Client(const Swift::JID& jid, const Swift::SafeString& password, Swift::NetworkFactories* networkFactories, Swift::Storages* storages) : 
+    Swift::Client(jid, password, networkFactories, storages),
+    mamResultParser("result", "urn:xmpp:mam:2", getPayloadParserFactories()),
+    mamFinParser("fin", "urn:xmpp:mam:2")
+{
+    addPayloadParserFactory(&mamResultParser);
+    addPayloadParserFactory(&mamFinParser);
+}
```

Comment:
Why do you need this class? Only for 'addPayloadParser'? Have a look here to get to know how to handle custom payload parsers:
https://swift.im/swiften/guide/

Reply:
The standard parsers are doing the work but they are not called because of the namespace value. I preferred to reuse them with a distinct namespace.

Reply:
Ok, got it. But it should be possible to instantiate them (mamResultPaser and mamFinParser) within e.g. the MamManager.cpp. Why do you need the Client.cpp class for that?

Reply:
Just made some tests. The problem here is that getPayloadParserFactories() is a protected member of Swift::Client. Only because of this you created the own client class. Looks like a clever solution in the first place. But if Swift::Client gets updated with new Swift releases, Shmoose still uses the own implementation. This can lead to an error.

Is it possible to derive from Swift::Client and do the extra initialization for the MaM stuff?